rectools/visuals/visual_app.py

```diff
@@ -273,6 +462,65 @@ def display(self) -> None:
             )
         )
 
+    def save(self, folder_name: str, overwrite: bool = False) -> None:
+        """Save stored data to re-create widgets when necessary. Use `VisualAppBase.load`
+        class method for re-creation or any other child classes (`VisualApp`, `ItemToItemVisualApp`).
+
+        Parameters
+        ----------
+        folder_name : str
+            Destination folder for data.
+        overwrite : bool, default ``False``
+            Allow to overwrite in the folder files if they already exist.
+        """
+        self.data_storage.save(folder_name, overwrite)
+
+    @classmethod
+    def load(
+        cls,
+        folder_name: str,
+        auto_display: bool = True,
+        formatters: tp.Optional[tp.Dict[str, tp.Callable]] = None,
+        rows_limit: int = 20,
+        min_width: int = 100,
+    ) -> "VisualAppBase":
+        """Create widgets from data that was saved earlier.
+
+        Parameters
+        ----------
+        folder_name : str
+            Destination folder for data.
+        auto_display : bool, optional, default ``True``
+            Display widgets right after initialization.
+        formatters : tp.Optional[tp.Dict[str, tp.Callable]], optional, default ``None``
+            Formatter functions to apply to columns elements in the sections of interactions and recos.
+            Keys of the dict must be columns names (item_data, interactions and recos columns can be
+            specified here). Values bust be functions that will be applied to corresponding columns
+            elements. The result of each function must be a unicode string that represents html code.
+            Formatters can be used to format text, create links and display images with html.
+        rows_limit : int, optional, default 20
+            Maximum number of rows to display in the sections of interactions and recos.
+        min_width : int, optional, default 100
+            Minimum column width in pixels for dataframe columns in widgets output. Must be greater then
+            10.
+
+        Returns
+        -------
+        VisualApp
+            Jupyter widgets for recommendations visualization.
+        """
+        data_storage = AppDataStorage.from_saved(folder_name=folder_name)
+
+        # todo: make different types of app based on data.storage.is_u2i?
+
+        return VisualAppBase(
```

Comment:
cls

Reply:
fixed

Reply:
Changed it back to `VisualAppBase`. Otherwise it doesn't work from VisualApp and ItemToItemVisualApp instances.
Btw, out python3.7 mypy version doesn't pass my classes inheritance structure with `_create_data_storage` method. But our version for python >=3.8 passes

Reply:
Its cls finally